js/packages/mobile-wallet-adapter-protocol/src/types.ts

```diff
@@ -63,6 +65,18 @@ export interface CloneAuthorizationAPI {
 export interface DeauthorizeAPI {
     deauthorize(params: { auth_token: AuthToken }): Promise<Readonly<Record<string, never>>>;
 }
+
+export interface GetCapabilitiesAPI {
+    getCapabilities(): Promise<
+        Readonly<{
+            supports_clone_authorization: boolean;
+            supports_sign_and_send_transactions: boolean;
+            max_transactions_per_request: boolean;
+            max_messages_per_request: boolean;
+            supported_transaction_versions: ReadonlyArray<TransactionVersion>;
```

Comment:
@jordansexton, anything you see particularly bad about this being a `ReadonlyArray` instead of a `Set`? I wanted to avoid this PR being anything other than `TypeScript`.

Reply:
We use ReadonlyArray in the standard too, I don't really care for Set either. I assume whatever uniqueness/perf benefit it might have is likely offset for small sets.

Reply:
In terms of implementation, I also prefer arrays since we want to enforce them being readonly, and cloning sets doesn't have a native API so we just end up spreading them to copy, which is probably bad for perf.